sdk/go/pulumi/context.go

```diff
@@ -1840,6 +1841,12 @@ func (ctx *Context) RegisterPackageRemoteComponentResource(
 	return ctx.registerResource(t, name, props, resource, true /*remote*/, packageRef, opts...)
 }
 
+func (ctx *Context) RegisterComponentResourceV2(
```

Comment:
I'd probably move this up iwth `RegisterComponentResource`, but that's very minor so feel free to ignore.